### PR TITLE
Updated Angles on Bracket arrow styles example to make angles clear #23176

### PR DESCRIPTION
**Pull request by jacoverster**
## PR Summary

PR relates to this [issue](https://github.com/matplotlib/matplotlib/issues/23176). Added `AngleAnnotation` patches to mark the angles and included `FancyArrowPatch` arrows to show the directions of AngleA and AngleB.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by timhoffm** [comment]
I wouldn't have a problem updating "what's new" in this case. This is "only" a visual improvement of an existing plot.

However, I agree that it's not very discoverable in there.

> Can you please make this an example in annotations and then link the API docs for arrows to it? @timhoffm may have a better suggestion but I'm not sure what the policy is in terms of these kinds of figures in the reference docs.

Fundamentally, illustrating plots are fine the reference docs, see e.g. the notes section of https://matplotlib.org/devdocs/api/ticker_api.html#matplotlib.ticker.ScalarFormatter. However the example code is quite long and would clutter the docstring. So, placing it in the text/annotations section of the examples and linking in the plot via `.. plot:: /path/to/example` at https://matplotlib.org/devdocs/api/_as_gen/matplotlib.patches.ArrowStyle.html#matplotlib.patches.ArrowStyle is the best solution here.

**Comment by jacoverster** [comment]
Hi @timhoffm and @story645 I've updated the PR with the proposed changes. I created a new example and linked it to matplotlib.patches.ArrowStyle in the example references. I also referenced the example in the ArrowStyle docstring. When I build the documentation it shows up as an example but it does not pull through to https://matplotlib.org/devdocs/api/_as_gen/matplotlib.patches.ArrowStyle.html#matplotlib.patches.ArrowStyle - I assume I'm missing something here so please assist with that.

**Comment by jklymak** [comment]
You've squashed some unrelated commits in here.  An easy way to fix is to copy the two files you wanted to change somewhere else, On your branch do `git fetch upstream; git reset --hard upstream/main` and then copy your two files back to where they belong and make your commit.   Alternately, you can do `git restore -s upstream/main pathTo/MyFile` for the two files.

**Comment by jacoverster** [comment]
Hi @jklymak, sorry about that wat trying to follow the process to [amend](https://matplotlib.org/stable/devel/coding_guide.html#summary-for-pull-request-authors) the previous commit. Did not notice the other commit.

**Comment by jacoverster** [comment]
> You've squashed some unrelated commits in here. An easy way to fix is to copy the two files you wanted to change somewhere else, On your branch do `git fetch upstream; git reset --hard upstream/main` and then copy your two files back to where they belong and make your commit. Alternately, you can do `git restore -s upstream/main pathTo/MyFile` for the two files.

Hi  @jklymak, not sure how that commit got entangled with mine. I tried this without avail, received the following:

```
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.
```

I went ahead and synced my branch with matplotlib:main from GitHub ("sync fork") and then pulled again but now it looks like some of the changes are already merged, so I'm a bit confused. Not sure what to do now.

Please assist and sorry for the mess.

**Comment by jklymak** [comment]
You need to add a "remote" called `upstream` before you can do that. Our git guide is quite bad, in my opinion.   This one is quite good (and I still think we should just copy it, and stop dragging our feet making a new one): https://docs.xarray.dev/en/stable/contributing.html#getting-started-with-git

**Comment by jacoverster** [comment]
Hi @jklymak, thanks I managed to set the upstream and then I ran:
```
git fetch upstream
git reset --hard upstream/main
```

Then made the changes again on my branch and ran:
```
git commit --amend --no-edit
git push --force-with-lease
```

I think it should be good now. Please confirm.

**Comment by jklymak** [comment]
Looks better.  Congrats!

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -3119,6 +3119,10 @@ class ArrowStyle(_Style):
     stroked. This is meant to be used to correct the location of the
     head so that it does not overshoot the destination point, but not all
     classes support it.
+
+    Examples
+    --------
+    .. plot:: gallery/text_labels_and_annotations/angles_on_bracket_arrows.py
```
[the remark this answers is not in the document]
Comment (jacoverster) [reply lib/matplotlib/patches.py:3125]: > Sorry I spaced - I think notes or examples is contextual and here this plot is a note about how angles work more than an example of how to use brackets

Got it, thanks. Will update tot a Note.

File: examples/text_labels_and_annotations/angles_on_bracket_arrows.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+
+#############################################################################
+#
+# .. admonition:: References
+#
+#    The use of the following functions, methods, classes and modules is shown
+#    in this example:
+#
+#    - `matplotlib.patches.ArrowStyle`
```
Comment (jacoverster) [inline examples/text_labels_and_annotations/angles_on_bracket_arrows.py:236]: I'm not sure why the mini gallery was not created from this? As per the [docs](https://matplotlib.org/devdocs/devel/documenting_mpl.html#references-for-sphinx-gallery) is this all that is required @story645 or it there some other reference required in the docstring?

File: lib/matplotlib/patches.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacoverster) [reply lib/matplotlib/patches.py]: > Also according to our docs, the mini gallery should have been registered as part of the references https://matplotlib.org/devdocs/devel/documenting_mpl.html#references-for-sphinx-gallery

Is anything else missing @story645? Any ideas why it is not registered?

File: lib/matplotlib/patches.py [hunk 3]
```diff
@@ -3119,6 +3119,10 @@ class ArrowStyle(_Style):
     stroked. This is meant to be used to correct the location of the
     head so that it does not overshoot the destination point, but not all
     classes support it.
+
+    Notes
+    -----
+    .. plot:: gallery/text_labels_and_annotations/angles_on_bracket_arrows.py
```
[the remark this answers is not in the document]
Comment (jacoverster) [reply lib/matplotlib/patches.py:3129]: If this is the expected behavior for all arrow styles then we can definitely simplify this a bit. Maybe we should open another issue and PR to keep the changes atomic? (And to see what happens to https://github.com/matplotlib/matplotlib/issues/24217)

**Comment by oscargus** [comment]
The failing test doesn't seem related to this PR:
```
/home/circleci/project/doc/api/prev_api_changes/api_changes_3.6.0/removals.rst:157: WARNING: py:obj reference target not found: SubplotBase.get_subplotspec
/home/circleci/project/doc/api/prev_api_changes/api_changes_3.6.0/removals.rst:158: WARNING: py:obj reference target not found: SubplotBase.set_subplotspec
```

**Comment by jacoverster** [comment]
> So I apologize profusely for doing this now, but I have an idea that might simplify things? Maybe instead of making this a new example (w/ all the annoyances of copying the ArcAnnotation class), this gets folded into the ArcAnnotation example as a second example? And then we title that example something like annotating angles - direction and scale invariance? B/C the direction stuff is what's really useful here as an example, and that the figure makes a good bracket reference is kind of independent...(and yes this then is maybe a tutorial, but that's out of scope here)
> 
> You can still keep the link to the figure in the reference using the sphinx gallery reference tags https://sphinx-gallery.github.io/stable/advanced.html#cross-referencing

Good suggestion. Another alternative could be to leave out AngleAnnotation completely (which basically just provides the angle size in this case) and only add the directional arrow. That's what we are really after from the start right, more clarity on the angle direction?

It would then look like this:

image

**Comment by jacoverster** [comment]
> > Alternatively, we could also leave out AngleAnnotation completely
> 
> I'd still like labeling of the angle between bracket and perpendicular, but that can probably be achieved w/ ax.text placed relative to the perpendicular.
> 
> Also I really like your suggestion!

Either one is cool, please let me know what the preferred way is 👌🏻

**Comment by jacoverster** [comment]
Ok, so I've simplified the example significantly by removing AngleAnnotation and adding ax.Text angles. The plot looks like this now: 
image

In the end it's exactly the same output as with AngleAnnotation, just a lot less code.

**Comment by jacoverster** [comment]
> build failed 'cause of unexpected indent but I'm having trouble finding it in the .py file
> 
> > /home/circleci/project/doc/gallery/text_labels_and_annotations/angles_on_bracket_arrows.rst:32: ERROR: Unexpected indentation.
> 
> eta: probably this line going by the rendered docs image

Ah, missed a sneaky space there. Should be fixed now. Sorry about that.

File: examples/text_labels_and_annotations/angles_on_bracket_arrows.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        # Define the connection directions for the annotation arrows
+        connection_dirs = (1, -1) if angle > 0 else (-1, 1)
+        # Add arrows and annotation text
+        arrowstyle = "Simple, tail_width=0.5, head_width=4, head_length=8"
+        for vline, dir, patch_top, angle in zip(vlines, connection_dirs,
+                                                patch_tops, anglesAB):
+            kw = dict(connectionstyle=f"arc3,rad={dir * 0.5}",
+                      arrowstyle=arrowstyle, color="C0")
+            ax.add_patch(FancyArrowPatch(vline, patch_top, **kw))
+            ax.text(vline[0] - dir * 0.15, y + 0.3, angle, ha="center",
```
Comment (QuLogic) [inline examples/text_labels_and_annotations/angles_on_bracket_arrows.py:54]: Small tweak:
```suggestion
            ax.text(vline[0] - dir * 0.15, y + 0.3, f'{angle}°', ha="center",
```

Comment (jacoverster) [reply examples/text_labels_and_annotations/angles_on_bracket_arrows.py:54]: Thanks for the suggestion. Will add it in.

**Comment by QuLogic** [comment]
Thanks @jacoverster! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Comment by jacoverster** [comment]
> Thanks @jacoverster! Congratulations on your first PR to Matplotlib 🎉 We hope to hear from you again.
It was a pleasure. Thanks for the guidance @story645 and @jklymak. Would love to contribute more.

**Pull request merged**